api/src/router/routes/github.js

```diff
@@ -22,10 +22,53 @@
 
 import { asyncMiddleware } from '@bcgov/common-nodejs-utils';
 import { Router } from 'express';
-import { createRecord, getRecords } from '../../libs/gh-utils/gh-ops';
+import {
+  updatePRState,
+  createRecord,
+  getRecords,
+  getRequestContent,
+} from '../../libs/gh-utils/gh-ops';
+import { getPR } from '../../libs/gh-utils/gh-requests';
+import { GITHUB_LABELS } from '../../constants';
 
 const router = new Router();
 
+/**
+ * Update record status:
+ * this is triggered after the KeyCloak job is done for IDIR and GitHub:
+ * 1. if success, tag ready-for-review, send notification
+ *   - if no BCeID label, then merge and close
+ *   - if BCeID, send notification to reviewer
+ * 2. if failed, tag request-failed and send notification with reason
+ * TODO: notification feature
+ */
+router.put(
+  '/records/setReady/:prNumber',
+  asyncMiddleware(async (req, res) => {
+    const { prNumber } = req.params;
+    const { keycloakJobResult } = req.body;
+    try {
+      const { status, message } = keycloakJobResult;
+      // get PR:
+      const pr = await getPR(prNumber);
+      // if failed, add label with reason:
+      if (status === 'failed') await updatePRState(pr, false, GITHUB_LABELS.FAILED, message);
+      // if BCeID, continue:
+      else if (pr.labels && pr.labels.includes(GITHUB_LABELS.BCEID))
+        await updatePRState(pr, false, GITHUB_LABELS.READY, message);
+      // if no BCeID merge and close:
+      else await updatePRState(pr, true, null, message);
+      res.status(204).end();
+    } catch (err) {
+      const errCode = err.status ? err.status : 500;
+      res.status(errCode).json(`Unable to update the PR ${prNumber}: ${err}`);
```

Comment:
does `.json` automatically interpret that argument to something like `{ message: 'unable to update...'}` ?

Reply:
that's what I thought but doesn't look like so, i've switch back to `send()`